apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/lru/LruCacheStore.java

```diff
@@ -54,7 +55,14 @@ public LruCacheStore(EvictionPolicy evictionPolicy, CacheStore secondaryCacheSto
       try {
         return lruCache.get(key, new Callable<Record>() {
           @Override public Record call() throws Exception {
-            return secondaryCacheStore.get().loadRecord(key);
+            Record record = secondaryCacheStore.get().loadRecord(key);
+            // get(key, callable) requires non-null. If null, an exception should be
+            //thrown, which will be converted to null in the catch clause.
+            if (record == null) {
+              throw new ExecutionException(
+                  new IOException(String.format("Record{id=%s} not present in secondary cache", key)));
```

Comment:
Hmm, yeah looks like the wrapping is unnecessary.

We could make `loadRecord` throw `IOException` if we feel that an `IOException` is better than returning `null` if the record is not present. I personally feel that `null` is fine. This would have to be on the `CacheStore` interface, so an exception seems a bit heavy handed to me.